langroid/agent/task.py

```diff
@@ -732,8 +733,37 @@ def run(
         if return_type is None:
             return_type = self.default_return_type
 
+        # If possible, take a final strict decoding step
+        # when the output does not match `return_type`
         if return_type is not None and return_type != ChatDocument:
-            return self.agent.from_ChatDocument(final_result, return_type)
+            parsed_result = self.agent.from_ChatDocument(final_result, return_type)
+
+            if (
+                parsed_result is None
+                and isinstance(self.agent, ChatAgent)
+                and self.agent._json_schema_available()
+            ):
+                strict_agent = self.agent[return_type]
+                output_args = strict_agent._function_args()[-1]
+                if output_args is not None:
```

Comment:
Need some comment here -- why does `output_args is not None` indicate  that the result did not adhere to the schema? Maybe becomes clear later but not immediately clear.

Reply:
This should be using `llm_response_async`; corrected (may have overwritten the line after some changes to the sync path)